spicelib/sim/simulator.py

```diff
@@ -136,6 +165,91 @@ def is_available(cls):
+    
+    @classmethod
+    def get_default_library_paths(cls) -> List[str]:
+        """
+        Return the directories that contain the standard simulator's libraries, 
+        as derived from the simulator's executable path and platform.
+        spice_exe must be set before calling this method.
+        
+        This is companion with `set_custom_library_paths()`
+
+        :return: the list of paths where the libraries should be located.
+        :rtype: List[str]
+        """
+        paths = []
+        myexe = None
+        # get the executable
+        if cls.spice_exe and len(cls.spice_exe) > 0:
+            # TODO: this will fail if the simulator executable is not in the last element of the list. Maybe make this more robust.
+            if os.path.exists(cls.spice_exe[-1]):            
+                myexe = cls.spice_exe[-1]
+        _logger.debug(f"Using Spice executable path '{myexe}' to determine the correct library paths.")
+        for path in cls._default_lib_paths:
+            _logger.debug(f"Checking if library path '{path}' exists.")
+            p = cls.expand_and_check_local_dir(path, myexe)
+            if p is not None:
+                _logger.debug(f"Adding path '{p}' to the library path list")
+                paths.append(p)
+        return paths
+    
+    @staticmethod
+    def expand_and_check_local_dir(path: str, exe_path: str = None) -> Optional[str]:
+        """
+        Expands a directory path to become an absolute path, while taking into account a potential use under wine (under MacOS and Linux). 
+        Will also check if that directory exists.
+        The path must either be an absolute path or start with ~. Relative paths are not supported.
+        On MacOS or Linux, it will try to replace any reference to the virtual windows root under wine into a host OS path.
+        
+        Examples:
+        * under windows:
+        * C:/mydir -> C:/mydir
+        * ~/mydir -> C:/Users/myuser/mydir
+        * under linux, and if the executable is /mywineroot/.wine/drive_c/(something):
+        * C:/mydir -> /mywineroot/.wine/drive_c/mydir
+        * ~/mydir -> /mywineroot/.wine/drive_c/users/myuser/mydir
+        
+        :param path: The path to expand. Must be in posix format, use `PureWindowsPath(path).as_posix()` to transform a windows path to a posix path.
+        :type path: str
+        :param exe_path: path to a related executable that may or may not be under wine, defaults to None, ignored on Windows
+        :type exe_path: str, optional
+        :return: the fully expanded path, as posix path, will return None if the path does not exist.
+        :rtype: Optional[str]
+        """
+        c_drive = None
+        if sys.platform == "linux" or sys.platform == "darwin":
+            if exe_path and "/drive_c/" in exe_path:
+                # this is very likely a wine path
+                c_drive = exe_path.split("/drive_c/")[0] + "/drive_c/"
+        if c_drive is not None:
```

Comment:
It seems that windows case is not covered here.

Reply:
windows is handled, more down, line 249 etc.

Reply:
made comments more explicit